finetuner/callbacks.py

```diff
@@ -0,0 +1,160 @@
+from dataclasses import dataclass, field
+from typing import Optional
+
+
+@dataclass
+class BestModelCheckpoint:
+    """
+    :param monitor: if `monitor='train_loss'` best model saved will be according
+        to the training loss, while if `monitor='val_loss'` best model saved will be
+        according to the validation loss.
+    :param mode: one of {'auto', 'min', 'max'}. The decision to overwrite the
+        currently saved model is made based on either the maximization or the
+        minimization of the monitored quantity.
+        For an evaluation metric, this should be `max`, for `val_loss` this should
+        be `min`, etc. In `auto` mode, the mode is set to `min` if `monitor='loss'`
+        or `monitor='val_loss'` and to `max` otherwise.
+    :param verbose: Whether to log notifications when a checkpoint is saved.
+    """
+
+    name: str = 'BestModelCheckpoint'
```

Comment:
you don't have to specify `name` on any callback. You can always get like this `callback.__class__.__name__`